src/server/modules/user/index.js

```diff
@@ -65,6 +66,60 @@ if (settings.user.auth.facebook.enabled) {
   );
 }
 
+if (settings.user.auth.google.enabled) {
+  passport.use(
+    new GoogleStrategy(
+      {
+        clientID: settings.user.auth.google.clientID,
+        clientSecret: settings.user.auth.google.clientSecret,
+        callbackURL: '/auth/google/callback'
+      },
+      async function(accessToken, refreshToken, profile, cb) {
+        const { id, username, displayName, emails: [{ value }] } = profile;
+        try {
+          let user = await User.getUserByGoogleIdOrEmail(id, value);
+
+          if (!user) {
+            const isActive = true;
+            const [createdUserId] = await User.register({
+              username: username ? username : value,
+              email: value,
+              password: id,
+              isActive
+            });
+
+            await User.createGoogleOuth({
+              id,
+              displayName,
+              userId: createdUserId
+            });
+
+            await User.editUserProfile({
```

Comment:
Setting the local user profile from the google profile information.

Should we do the same for Facebook?

Reply:
@verdverm Yes, would be good, if we can access this data freely from Facebook without asking permission from user.

Reply:
The scopes (per oauth provider) determine what the user sees and agrees to give access to.

There are probably two sensible defaults in general, `email only` and `email + basic profile`.  Then people who use this project can add configuration to expand the scopes they need. We could provide a hook in the auth cycle to extract and store extra info during registration too. Many oauth permission grants are used with apis and I would expect this is more application dependent.